### PR TITLE
[#4369] Added new query_processor library (master)

### PR DESCRIPTION
**Pull request by korydraughn**
Also adds a new error code for situations where the error is unknown.

[Passed CI](http://172.25.14.63:8080/view/Personal/job/irods-build-and-test-workflow/1998/)

### CODE REVIEW

**Comment by trel** [comment]
seems good to me, let's # and get it merged - would be nice to have in 4.2.6, and won't hurt anything since we aren't wiring it up for use, yet.

File: cmake/development_library.cmake [hunk 1]
```diff
@@ -87,6 +87,7 @@ set(
   ${CMAKE_SOURCE_DIR}/lib/core/include/irods_tcp_object.hpp
   ${CMAKE_SOURCE_DIR}/lib/core/include/irods_threads.hpp
   ${CMAKE_SOURCE_DIR}/lib/core/include/irods_virtual_path.hpp
+  ${CMAKE_SOURCE_DIR}/lib/core/include/query_processor.hpp
```
Comment (alanking) [inline cmake/development_library.cmake]: Nit: I haven't audited this list but alphabetization would be nice here.

File: lib/core/include/rodsErrorTable.h [hunk 2]
```diff
@@ -233,6 +233,7 @@ NEW_ERROR(SYS_SOCK_WRITE_ERR,                          -161000)
 NEW_ERROR(SYS_SOCK_CONNECT_ERR,                        -162000)
 NEW_ERROR(SYS_OPERATION_IN_PROGRESS,                   -163000)
 NEW_ERROR(SYS_REPLICA_DOES_NOT_EXIST,                  -164000)
+NEW_ERROR(SYS_UNKNOWN_ERROR,                           -165000)
```
Comment (alanking) [inline lib/core/include/rodsErrorTable.h]: `SYS_INTERNAL_ERR` may communicate the same amount of information if we want to avoid adding an new error.

Comment (korydraughn) [reply lib/core/include/rodsErrorTable.h]: This new error code was introduced during a group code review for cases where devs choose to use -1 to indicate an error occurred, but aren't sure what happened.

**Comment by korydraughn** [comment]
Added #.

**Pull request merged**